sshconnect2.c

```diff
@@ -255,14 +256,19 @@ ssh_kex2(struct ssh *ssh, char *host, struct sockaddr *hostaddr, u_short port,
 	    myproposal[PROPOSAL_MAC_ALGS_STOC] = options.macs;
 	if (use_known_hosts_order) {
 		/* Query known_hosts and prefer algorithms that appear there */
-		myproposal[PROPOSAL_SERVER_HOST_KEY_ALGS] = prop_hostkey =
-		    compat_pkalg_proposal(ssh,
-		    order_hostkeyalgs(host, hostaddr, port, cinfo));
+		if ((hostkeyalgs = order_hostkeyalgs(host, hostaddr, port, cinfo)) == NULL)
+			fatal_f("order_hostkeyalgs");
+		pkalg = match_filter_allowlist(hostkeyalgs, options.pubkey_accepted_algos);
```

Comment:
This isn't correct: `options.pubkey_accepted_algos` is the `PubkeyAcceptedAlgorithms` configuration option. That is for controlling which signature algorithms are used for user authentication, whereas this is host authentication.

The configuration option that you're looking for is `HostkeyAlgorithms`, but this is already applied inside `order_hostkeyalgs()`